ipatests/test_integration/test_subids.py

```diff
@@ -268,3 +280,25 @@ def test_sudid_match_shows_uid(self):
                                           f"--subuid={subuid}"])
         owner = self._parse_result(result)["owner"]
         assert owner == uid
+
+    def test_nsswitch_is_updated_with_subid_entry(self):
+        """
+        This test case checks that when ipa-client-install
+        is installed with --subid option, the nsswitch.conf
+        file is modified with the entry 'subid: sss'
+        """
+        cmd = self.clients[0].run_command(
+            ["grep", "^subid", "/etc/nsswitch.conf"]
+        )
+        subid = cmd.stdout_text.split()
+        assert ['subid:', 'sss'] == subid
+
+    def test_nsswitch_doesnot_contain_subid_entry(self):
+        """
+        This testcase checks that when ipa-client-install
+        is installed without subid option, the nsswitch.conf
+        doesnot contain subid entry
+        """
+        tasks.uninstall_client(self.clients[0])
+        tasks.install_client(self.master, self.clients[0])
+        assert 'subid' not in /etc/nsswitch.conf
```

Comment:
This line is not python-compliant and fails in pylint / in the build.

Apart from the syntax, I have multiple comments:
- you can optimize the number of times the client is installed/uninstalled and simplify the code. To do so:
  - remove the install method from this class. This means the standard installation method from IntegrationTest will be used instead, without the --subid option for the client
  - move this test before `test_nsswitch_is_updated_with_subid_entry`, and remove the uninstall/install calls from this test.
  - in `test_nsswitch_is_updated_with_subid_entry`, add calls to uninstall/install with the extra arg --subid
- In this test, we want to make sure that either `subid:` is not defined at all or if present, it does not contain sss as source. Your code needs to read the content of /etc/nsswitch.conf **on the client**. It can run a grep command with "^subid:" but must succeed if grep doesn't return anything or returns a line that doesn't contains sss.